src/java/com/puppetlabs/ssl_utils/SSLUtils.java

```diff
@@ -298,6 +298,39 @@ public static boolean isRevoked(X509CRL crl, X509Certificate certificate) {
         return crl.isRevoked(certificate);
     }
 
+    private static X509v2CRLBuilder crlBuilder(X509CRL crl)
+            throws CRLException {
+        // The CRL is not valid if the time of checking == the time of last_update.
+        // So to have it valid right now we need to say that it was updated one second ago.
+        DateTime now = DateTime.now();
+        Date thisUpdate = now.minusSeconds(1).toDate();
+        Date nextUpdate = now.plusYears(5).toDate();
+        X509v2CRLBuilder builder =
+                new JcaX509v2CRLBuilder(crl.getIssuerX500Principal(), thisUpdate);
+        builder.setNextUpdate(nextUpdate);
+        // Copy over existing CRLEntrys
+        builder.addCRL(new JcaX509CRLHolder(crl));
+        return builder;
+    }
+
+    private static X509CRL buildCRL(X509CRL crl,
+                                    PrivateKey issuerPrivateKey,
+                                    PublicKey issuerPublicKey,
+                                    X509v2CRLBuilder builder)
+            throws IOException, CRLException, NoSuchAlgorithmException, OperatorCreationException {
+        BigInteger crlNumber = (BigInteger)
```

Comment:
What is this doing? Is this, like, a type coercion or something? or is it like `maybe`?

Reply:
It's casting. So yeah, like coercion.